agent/pkg/builder/container/utils.go

```diff
@@ -44,6 +44,48 @@ func pullImage(ctx context.Context, logger io.StringWriter, fullyQualifiedImageN
 			}
 		}
 
+		return nil, err
+	}
+
+	return cli, nil
+}
+
+func imageExists(ctx context.Context, logger io.StringWriter, fullyQualifiedImageName string) (bool, error) {
+	cli, err := createCli(logger)
+	if cli == nil {
+		return false, err
+	}
+
+	filter := filters.NewArgs()
+	filter.Add("reference", fullyQualifiedImageName)
+
+	images, err := cli.ImageList(ctx, types.ImageListOptions{Filters: filter})
+	if err != nil {
+		if logger != nil {
+			_, err = logger.WriteString("Failed to list images")
+			if err != nil {
+				fmt.Printf("Failed to write log: %s", err.Error())
+			}
+		}
+
+		return false, err
+	}
+
+	if count := len(images); count == 1 {
+		return true, nil
+	} else if count > 1 {
+		return false, errors.New("unexpected image count")
+	}
+
+	return false, nil
+}
+
+// force pulls the given image name
+func pullImage(logger io.StringWriter, fullyQualifiedImageName, authCreds string) error {
+	ctx := context.Background()
```

Comment:
Pulling an image being a long process, it would be good to handle context cancellation, parent contexts.